lib/applications/resolvers.ts

```diff
@@ -611,3 +613,93 @@ export const createReplacementApplication: Resolver = async (_, args, { prisma }
     ok: true,
   };
 };
+
+export const generateApplicantsReport: Resolver = async (_, args, { prisma }) => {
+  const {
+    input: { startDate, endDate, columns },
+  } = args;
+
+  const columnsSet = new Set(columns);
+
+  const applications: GenerateApplicantsReportApplications[] = await prisma.application.findMany({
+    where: {
+      createdAt: {
+        gte: new Date(startDate),
+        lte: new Date(endDate),
+      },
+    },
+    select: {
+      rcdUserId: columnsSet.has(ApplicationsReportColumn.UserId),
+      firstName: columnsSet.has(ApplicationsReportColumn.ApplicantName),
+      middleName: columnsSet.has(ApplicationsReportColumn.ApplicantName),
+      lastName: columnsSet.has(ApplicationsReportColumn.ApplicantName),
+      dateOfBirth: columnsSet.has(ApplicationsReportColumn.ApplicantDob),
+      createdAt: columnsSet.has(ApplicationsReportColumn.ApplicationDate),
+      paymentMethod: columnsSet.has(ApplicationsReportColumn.PaymentMethod),
+      processingFee:
+        columnsSet.has(ApplicationsReportColumn.FeeAmount) ||
+        columnsSet.has(ApplicationsReportColumn.TotalAmount),
+      donationAmount:
+        columnsSet.has(ApplicationsReportColumn.DonationAmount) ||
+        columnsSet.has(ApplicationsReportColumn.TotalAmount),
+      permits: {
+        select: {
+          rcdPermitId: true,
+        },
+      },
+    },
+  });
+
+  // Adds totalAmount, applicantName and rcdPermitId properties to applications object
+  applications.map(application => {
```

Comment:
Here we can make a new variable that is a result of this map so that we don't need to modify the original data. This requires more space but is easier to maintain and eliminates TypeScript errors

Reply:
Good suggestion! Done!